src/pushcollector/_impl/proxy.py

```diff
@@ -50,11 +51,48 @@ class CollectorProxy(object):
     def __init__(self, delegate):
         self._delegate = delegate
 
+    def _translate_pushitem(self, pushitem):
+        def _get_checksum():
+            if pushitem.md5sum and pushitem.sha256sum:
+                return {"md5": pushitem.md5sum, "sha256": pushitem.sha256sum}
+            else:
+                return None
+
+        pushitem = {
+            "filename": pushitem.name,
+            "state": pushitem.state,
+            "src": pushitem.src,
+            "dest": str(pushitem.dest) if pushitem.dest else None,
```

Comment:
This doesn't look correct to me, two problems:

- if it's right to serialize a list into a string like this, I don't think python str() would be the right way... I don't think there's any guarantee about that format.
- it won't give a correct result anyway to collapse the list of destinations into a single string here.  The way the push items are stored in Pub today is for this field to end up containing a single destination only.  If you have a look in `log_push_items` method in Pub, you can see there's a loops where it takes input items which may contain a *list* of destinations, and ends up producing one output push item for each destination on each input item.  I think the same logic is needed here.

So it seems like `_translate_pushitem(self, pushitem)` should be defined as returning a *list of* dicts rather than just one dict.

Reply:
I wasn't sure on this part whether there should be a push item for each destination or one push item with the destination as it's just collecting the push info like logs. Probably should have put a comment here earlier.